otx/api/entities/datasets.py

```diff
@@ -82,7 +82,10 @@ def __next__(self) -> DatasetItemEntity:
         return item
 
 
-class DatasetEntity:
+TDatasetItemEntity = TypeVar("TDatasetItemEntity", bound="DatasetItemEntity")
+
+
+class DatasetEntity(Generic[TDatasetItemEntity]):
```

Comment:
I also concerned about it and this is the reason why I newly created `DatasetItemEntityWithID` rather than adding `id` field to `DatasetItemEntity`. So, the changes in this PR were limited to the typing, not the code execution parts.

Please check that this change does not affect to Geti:
1. https://github.com/intel-innersource/frameworks.ai.interactive-ai-workflow.sonoma-creek/pull/7291
2. http://validationreports.sclab.intel.com:8004/reports/build_number_report?test_session_build_number=IMPTOps-59268&environment=mclx77

The one failed case in the `gui_on_merge_request` is not related to this PR.

Reply:
However, I recommend to include `id` field in `DatasetItemEntity` in the future. This is because Datumaro has `id` field on it's  `DatasetItem` object, so that it is needed for the compatibility. In addition, we have a use-case for the `id` field in noisy label detection feature.